File: python/binary-search/binary_search.py
def binary_search(list_of_numbers, number):
  if len(list_of_numbers) == 0:
    raise ValueError('Not found')
  min = 0
  max = len(list_of_numbers) - 1
  while min <= max:
    mid = (min + max) // 2
    if list_of_numbers[mid] == number:
      return mid
    if list_of_numbers[mid] > number:
      max = mid - 1
    else:
      min = mid + 1
  
  raise ValueError('Not found')

File: java/binary-search/src/main/java/BinarySearch.java
import java.util.List;

class BinarySearch {
  private List<Integer> list;

  BinarySearch(List<Integer> list) {
    this.list = list;
  }

  int indexOf(int target) throws ValueNotFoundException {
    int min = 0;
    int max = list.size() - 1;
    while (min <= max) {
      int mid = (min + max) / 2;
      int midValue = list.get(mid);
      if (midValue == target) {
        return mid;
      }
      if (target < midValue) {
        max = mid - 1;
      } else {
        min = mid + 1;
      }
    }
    throw new ValueNotFoundException("Value not in array");
  }
}